src/network/networkd-neighbor.h

```diff
@@ -22,8 +22,12 @@ struct Neighbor {
 
         int family;
         union in_addr_union in_addr;
-        bool mac_configured;
-        struct ether_addr mac;
+        union {
+                struct ether_addr mac;
+                union in_addr_union ip;
+        } lladdr;
+        bool use_mac:1;
+        bool use_ip:1;
```

Comment:
hmm, these two can never be on, can they?

doesn't matter much, but maybe make this an enum of its own?

```c
enum {
        NEIGHBOR_MAC,
        NEIGHBOR_IP,
        _NEIGHBOR_MAX,
        _NEIGHBOR_INVALID = -1,
};
```

Reply:
Done.